modules/appeals_api/app/models/appeals_api/supplemental_claim.rb

```diff
@@ -55,6 +57,16 @@ def pdf_structure(pdf_version)
       ).new(self)
     end
 
+    def assign_metadata
+      return unless api_version&.downcase == 'v2'
+
+      self.metadata = if Flipper.enabled?(:decision_review_sc_pact_act_boolean)
+                        { form_data: { evidence_type:, potential_pact_act: }, pact: { potential_pact_act: } }
```

Comment:
I feel like I'm missing some fancy rails-ism here, because all I see is a hash full of nils and wondering how your "not nil" specs are passing. I suspect adding the methods that are the same as the  keys makes sure those values get filled out automagically?

Reply:
That's exactly what it is. It was new to me, too! It was a requirement from Rubocop since the method names matched the keys.

Reply:
Realized I should also call out here that `evidence_type` has moved from the top-level of the metadata to under `form_data`. If you are good with this change, it will require a data migration for the Supplemental Claims records that already exist in Production.

Reply:
Yup this is fine by me!

Reply:
Same, i hit that in rubo a few days ago also, not really thrilled...but whatever, no big deal....

Reply:
Oooh this is that new fancy good good from Ruby 3. Destructuring and "restructuring". It feels nice to me, though still new and foreign.